pkg/controller/buildrun/buildrun_controller.go

```diff
@@ -369,11 +369,48 @@ func (r *ReconcileBuildRun) Reconcile(request reconcile.Request) (reconcile.Resu
 
 				// Increase BuildRun count in metrics
 				buildmetrics.BuildRunCountInc(buildRun.Status.BuildSpec.StrategyRef.Name)
-				// Add BuildRun establish and completion times in metrics
-				buildRunCompletionDuring := buildRun.Status.CompletionTime.Time.Sub(buildRun.CreationTimestamp.Time)
-				buildRunEstablishDuring := buildRun.Status.StartTime.Time.Sub(buildRun.CreationTimestamp.Time)
-				buildmetrics.BuildRunEstablishObserve(buildRun.Status.BuildSpec.StrategyRef.Name, buildRun.Namespace, buildRunEstablishDuring)
-				buildmetrics.BuildRunCompletionObserve(buildRun.Status.BuildSpec.StrategyRef.Name, buildRun.Namespace, buildRunCompletionDuring)
+
+				// Time between the creation of the buildrun and the start of the buildrun (buildrun ramp-up)
+				buildmetrics.BuildRunEstablishObserve(
+					buildRun.Status.BuildSpec.StrategyRef.Name,
+					buildRun.Namespace,
+					buildRun.Status.StartTime.Time.Sub(buildRun.CreationTimestamp.Time),
+				)
+
+				// Total time between the creation of the buildrun and the buildrun completion
+				buildmetrics.BuildRunCompletionObserve(
+					buildRun.Status.BuildSpec.StrategyRef.Name,
+					buildRun.Namespace,
+					buildRun.Status.CompletionTime.Time.Sub(buildRun.CreationTimestamp.Time),
+				)
+
+				// Look for the pod created by the taskrun
+				var pod = &corev1.Pod{}
+				if err := r.client.Get(ctx, types.NamespacedName{Namespace: request.Namespace, Name: lastTaskRun.Status.PodName}, pod); err == nil {
+					lastInitPodIdx := len(pod.Status.InitContainerStatuses) - 1
+					lastInitPod := pod.Status.InitContainerStatuses[lastInitPodIdx]
+
+					// buildrun ramp-up duration (time between buildrun creation and taskrun creation)
+					buildmetrics.BuildRunRampUpDurationObserve(
+						buildRun.Status.BuildSpec.StrategyRef.Name,
+						buildRun.Namespace,
+						lastTaskRun.CreationTimestamp.Time.Sub(buildRun.CreationTimestamp.Time),
+					)
```

Comment:
This block can go out of the `if` block. Sry, overlooked that yesterday.

Reply:
I have a look.